aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -17,23 +17,221 @@
 #include <cmath>
 
 namespace at { namespace native {
+static inline Tensor _fft(
+    const Tensor &self, int64_t signal_ndim, bool complex_input,
+    const bool complex_output, bool inverse, IntArrayRef signal_sizes,
+    fft_norm_mode normalization, bool onesided);
+
+namespace {
+
+ScalarType promote_type_fft(ScalarType type, bool require_complex) {
+  if (at::isComplexType(type)) {
+    return type;
+  }
+  // Promote integral to default float type
+  if (!at::isFloatingType(type)) {
+    type = c10::typeMetaToScalarType(c10::get_default_dtype());
+  }
+
+  TORCH_CHECK(type == kFloat || type == kDouble, "Unsupported dtype ", type);
```

Comment:
correct me if I am wrong but anytime the input type is not complex, it will just error out, right? if so, the code after this check will never be executed ...